README.md

```diff
@@ -76,317 +110,607 @@ cartridge --help
 
-# The main purpose of this script is to remove build artifacts from result package.
-# It will be ran after `tarantoolctl rocks make` on application build
+If you specify an existing directory in the `CARTRIDGE_BUILDDIR` environment
+variable, the `CARTRIDGE_BUILDDIR/build.cartridge` repository will be used for
+build and then removed. This directory will be cleaned up before building the
+application.
 
-rm -rf third_party
-rm -rf node_modules
-rm -rf doc
-```
+  **NOTE:** The specified directory cannot be an application subdirectory.
+
+#### Distribution directory
+
+For each distribution type, a temporary directory with application source files
+is created (further on we address it as *application directory*).
+This includes 3 stages.
+
+##### Stage 1. Cleaning up the application directory
+
+On this stage, some files are filtered out of the application directory:
+
+* First, `git clean -X -d -f` removes all untracked and
+  ignored files (it works for submodules, too).
+* After that, `.rocks` and `.git` directories are removed.
+
+Files permissions are preserved, and the code files owner is set to
+`root:root` in the resulting package.
+
+  **Note**: All application files should have at least `a+r` permissions
+  (`a+rx` for directories).
+  Otherwise, `cartridge pack` command raises an error.
+
+##### Stage 2. Building the application
+
+On this stage, `cartridge` [builds](#building-an-application) the application
+in the cleaned up application directory.
+
+##### Stage 3. Cleaning up the files before packing
+
+On this stage, `cartridge` runs `cartridge.post-build` (if it exists) to remove
+junk files (like `node_modules`) generated during application build.
 
-## Application packing type-specific details
+See an [example](#example-cartridgepost-build) in [special files](#special-files).
 
 ### TGZ
 
-`cartridge pack tgz ./myapp` will create a .tgz archive containing the application
-source code and rocks modules described in the application rockspec.
+`cartridge pack tgz ./myapp` creates a .tgz archive. It contains all files from the
+[distribution directory](#distribution-directory)
+(i.e. the application source code and rocks modules described in the application
+rockspec).
+
+The package name is `<name>-<version>.tar.gz`.
 
 ### RPM and DEB
 
-`cartridge pack rpm|deb ./myapp` will create an RPM or DEB package.
+`cartridge pack rpm|deb ./myapp` creates an RPM or DEB package.
+
+The package name is `<name>-<version>.{rpm,deb}`.
+
+It contains meta information: the package name (which is the application name),
+and the package version.
 
 If you use an opensource version of Tarantool, the package has a `tarantool`
 dependency (version >= `<major>.<minor>` and < `<major+1>`, where `<major>.<minor>`
-is the version of Tarantool used for application packing).
+is the version of Tarantool used for packing the application).
 You should enable the Tarantool repo to allow your package manager install
```

Comment:
Before installing the package, you should ...

I think, it can confuse (I decided that I should do it before packing).